### DISCUSSION TITLE
Editor margin shows up outside the editor

### DISCUSSION THREAD

**Discussion by nvmkpk** [VsixCommunity/Community.VisualStudio.Toolkit#265] [Q&A] [answered]
I created a custom editor margin. It all works fine except for one issue. It also appears in the expression editor of watch and quick watch window. Ho do I ensure my margin is visible inside full text editor only?

**Comment by reduckted** [comment, marked as answer]
It sounds like you're missing some attributes on your `IWpfTextViewMarginProvider`.

If you add the [`ContentType`](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.utilities.contenttypeattribute) attribute and the [`TextViewRole`](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.text.editor.textviewroleattribute) attribute, then you should be able to get your margin to only appear on editors.

```csharp
[Export(typeof(IWpfTextViewMarginProvider))]
[Name(MyMargin.Name)]
[ContentType("text")]
[TextViewRole(PredefinedTextViewRoles.Document)]
public class MyMargin : IWpfTextViewMarginProvider {
    // ...
}
```

There are different types of roles that you can use. You can see the pre-defined roles here: https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.text.editor.predefinedtextviewroles

**Reply by nvmkpk** [reply]
I already have both of those attriutes. `TextViewRoleAttribute` had `PredefinedTextViewRoles.Interactive`. After posting this question, I tried a few other members of `PredefinedTextViewRoles` but somehow I missed `PredefinedTextViewRoles.Document`. This fixed the issue,  thank you.
